scipy/fft/_pocketfft/pocketfft_hdronly.h

```diff
@@ -105,7 +110,7 @@ constexpr bool FORWARD  = true,
 #if defined(__INTEL_COMPILER)
 // do nothing. This is necessary because this compiler also sets __GNUC__.
 #elif defined(__clang__)
-#if __clang__>=5
+#if __clang_major__>=5
```

Comment:
cc @mreineck this is broken on AppleClang since it gives the XCode version here instead of the base clang version.

Reply:
This is a problem, because I noticed that the original version was broken everywhere else ...
On Linux with clang++ 9, `__clang__` is defined as 1 and `__clang_major__` is 9.
Do we need another level of `#ifdef`s here?

Reply:
I think you can just guard it with `#ifdef __APPLE__`. 

According to [wikipedia](https://en.wikipedia.org/wiki/Xcode#Toolchain_versions) clang 5 corresponds to Xcode 9.3 and above. So, it would be 
```cpp
#ifdef __APPLE__
#  if (__clang_major__ > 9) || (__clang_major__ == 9 && __clang_minor__ >= 3) 
#     undef POCKETFFT_NO_VECTORS
#  endif
#elif __clang_major__ >= 5
#  undef POCKETFFT_NO_VECTORS
#endif
```

Reply:
This is fine! I'll apply this patch upstream, once CI passes.